0.52.0 breaks cross compilation of sys dependencies

The new [0.52.0](https://github.com/bazelbuild/rules_rust/releases/tag/0.52.0) release breaks my build completely.

The issue seems to be related to vendored sys-crates not compiling the underlying c deps anymore.
So far I was able to pinpoint the problem to the following crates:
* mimalloc
* lz4-sys

The only viable workaround is to pin rules_rust to any of the previous versions. 

I made a repro of the issue that builds with the previous 0.51 version and virtually all before, but when using the new 0.52 version, it throws the exact same error that I see in my main mono repo.

Link to the repro: 
https://github.com/marvin-hansen/sys-crates-repro/tree/main

Here you go.

Sample error. Full log attached.

Depending on which sys crate gets compiled first, the error message may vary, 
but it is always some C import that was not found. 

```bash
cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
CC_SHELL_ESCAPED_FLAGS = None
cargo:warning=liblz4/lib/lz4.c:224:11: fatal error: 'stdlib.h' file not found
cargo:warning=  224 | # include <stdlib.h>   /* malloc, calloc, free */
cargo:warning=      |           ^~~~~~~~~~
cargo:warning=1 error generated.

--stderr:
```

[error-log.txt](https://github.com/user-attachments/files/17260953/error-log.txt)

> Here you go.
> 
> Sample error. Full log attached.
> 
> Depending on which sys crate gets compiled first, the error message may vary, but it is always some C import that was not found.
> 
> ```shell
> cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
> CC_SHELL_ESCAPED_FLAGS = None
> cargo:warning=liblz4/lib/lz4.c:224:11: fatal error: 'stdlib.h' file not found
> cargo:warning=  224 | # include <stdlib.h>   /* malloc, calloc, free */
> cargo:warning=      |           ^~~~~~~~~~
> cargo:warning=1 error generated.
> 
> --stderr:
> ```
> 
> [error-log.txt](https://github.com/user-attachments/files/17260953/error-log.txt)

So if I understand this log correctly,  is_absolute failed on "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.0.sdk"

Does this dir exists?

If no, then the issue somewhere there, here we exposing that one.

--- a/cargo/private/cargo_build_script.bzl
+++ b/cargo/private/cargo_build_script.bzl
@@ -182,7 +182,7 @@ def _pwd_flags(args):
             fix_next_arg = False
         else:
             opt, path = split_flag(arg)
-            if opt and path:
+            if opt and path and not paths.is_absolute(path):
                 res.append("{}${{pwd}}/{}".format(opt, path))
             else:
                 fix_next_arg = (opt != None)

